src/utils.js

```diff
@@ -47,6 +47,122 @@ module.exports = {
+    const osInfo = tln.getOsInfo();
+    const platform = osInfo.platform;
+    //
+    let cp_cmd = 'cp';
+    let cat_cmd = 'cat';
+    if (platform === 'win32') {
+      cp_cmd = 'copy';
+      cat_cmd = 'type';
+    }
+    let arr = [`${cp_cmd} ${src} ${dest}`];
+    for (let i of tail) {
+      arr.push(`${cat_cmd} ${i} >> ${dest}`);
+    }
+    script.set(arr);
+  },
+  getDefaultScript: (tln, id, distrs, script) => {
+    let s = [];
+    if (distrs[id]) {
+      Object.keys(distrs[id]).forEach( k => {
+        if (tln.filter.validate(k)) {
+          s = distrs[id][k];
+        }
+      });
+    }
+    if (s.length) {
+      script.set(s);
+      return true;
+    }
+    return false;
+  },
+  canInstallComponent: (tln, script) => {
+    const emptyDir = require('empty-dir');
+    //
+    const home = script.env['COMPONENT_HOME'];
+    const id = script.env['COMPONENT_ID'];
+    if (emptyDir.sync(home)) {
+      return true;
+    }
+    script.set([
+      `echo Component '${id}' is already installed at this location '${home}'`
+    ]);
+    return false;
+  },
+  getDownloadScriptById: (tln, id, distrs) => {
+    if (distrs[id]) {
+      return tln.utils.getDownloadScript(tln, distrs[id]);
+    }
+  },
+  getDownloadScript: (tln, dist) => {
+    let r = [];
+    const osInfo = tln.getOsInfo();
+    const platform = osInfo.platform;
+    if (dist[platform]) {
+      const opts = dist[platform].opts;
+      const name = dist[platform].name;
+      const url = dist[platform].url;
+      //
+      if (platform === 'win32') {
+        r.push(`echo Downloading ${url}`);
+        r.push(`powershell -Command "(New-Object System.Net.WebClient).DownloadFile('${url}', '${name}')"`);
+        if (name.match('tar.gz')) {
+          r.push(`tar -xvzf ${name}`);
+        } else {
+          r.push(`powershell -Command "Expand-Archive -LiteralPath '${name}' -DestinationPath '.'"`);
+        }
+        // move content
+        if (opts) {
+          if (opts.src && opts.flt && opts.dest) {
+            r.push(`powershell -Command "Get-ChildItem -Path '${opts.src}' -Filter '${opts.flt}' -Recurse | Move-Item -Destination '${opts.dest}'"`);
+            if (opts.rmv) {
+              r.push(`powershell -Command "Remove-Item '${opts.rmv}'"`);
+            }
+          }
+        }
+        r.push(`powershell -Command "Remove-Item '${name}'"`);
+      } else if (platform === 'linux') {
+        r.push(`wget '${url}'`);
+        if (name.match('tar.gz')) {
+          r.push(`tar -xzf '${name}'`);
+        } else {
+          r.push(`unzip '${name}'`);
+        }
+        // move content
+        if (opts) {
+          if (opts.src && opts.flt && opts.dest) {
+            r.push(`mv ${opts.src}/${opts.flt} ${opts.dest}`);
+            if (opts.rmv) {
+              r.push(`rmdir '${opts.rmv}'`);
+            }
+          }
+        }
+        r.push(`rm -f ${name}`);
+      } else if (platform === 'darwin') {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either merge this branch with the identical one on line "130" or change one of the implementations. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://159.65.230.204:9000/coding_rules#rule_key=javascript%3AS1871)